src/components/Personalization/createApplyPropositions.js

```diff
@@ -12,61 +12,78 @@ governing permissions and limitations under the License.
-        item.data.selector =
-          metadataForScope.selector || DEFAULT_METADATA.selector;
-        item.data.type = metadataForScope.actionType || DEFAULT_METADATA.type;
-      }
-    });
+export default ({ executeDecisions }) => {
+  const filterItemsPredicate = item => {
+    if (SUPPORTED_SCHEMAS.includes(item.schema)) {
+      return true;
+    }
+    return false;
+  };
+
+  const updatePropositionItems = ({ proposition, metadataForScope }) => {
+    proposition.items = proposition.items
+      .filter(filterItemsPredicate)
+      .map(item => {
+        if (item.schema === HTML_CONTENT_ITEM && isObject(metadataForScope)) {
+          item.data.selector = metadataForScope.selector;
+          item.data.type = metadataForScope.actionType;
+        }
+        return item;
+      });
     return proposition;
   };
 
+  const filterPropositionsPredicate = proposition => {
+    if (proposition.scope === PAGE_WIDE_SCOPE && proposition.renderAttempted) {
+      return false;
+    }
+
+    return true;
+  };
+
   const preparePropositions = ({ propositions, metadata }) => {
-    return Promise.resolve(
-      propositions.map(proposition => {
-        const completeProposition = { ...proposition };
+    return propositions
+      .filter(filterPropositionsPredicate)
+      .map(proposition => {
+        const completeProposition = (({ id, items, scope, scopeDetails }) => ({
+          id,
+          items,
+          scope,
+          scopeDetails
+        }))(proposition);
         if (isNonEmptyArray(completeProposition.items)) {
-          const metadataForScope =
-            !isEmptyObject(metadata) &&
-            isObject(metadata[completeProposition.scope])
-              ? metadata[completeProposition.scope]
-              : DEFAULT_METADATA;
-          return updatePropositions({
+          return updatePropositionItems({
             proposition: completeProposition,
-            metadataForScope
+            metadataForScope: metadata[completeProposition.scope]
           });
         }
         return completeProposition;
       })
-    );
+      .filter(proposition => isNonEmptyArray(proposition.items));
   };
 
   const applyPropositions = ({ propositions, metadata }) => {
-    return preparePropositions({ propositions, metadata })
-      .then(completePropositions => executeDecisions(completePropositions))
-      .then(() => {
-        showContainers();
-        propositions.forEach(proposition => delete proposition.renderAttempted);
-        return composePersonalizationResultingObject(propositions, true);
-      });
+    const propositionsToExecute = preparePropositions({
+      propositions,
+      metadata
+    });
+    return executeDecisions(propositionsToExecute).then(() => {
+      return Promise.resolve(
+        composePersonalizationResultingObject(propositionsToExecute, true)
+      );
+    });
   };
 
   return ({ propositions, metadata = {} }) => {
     if (isNonEmptyArray(propositions)) {
```

Comment:
Is this check necessary? The parameter validation should make sure that propositions is an array. If it is an empty array, then executeDecisions won't do anything anyways. Then you could just return the applyPropositions function.